File: _boj_all/2747.cpp
#include <bits/stdc++.h>
#define fastio cin.tie(0)->sync_with_stdio(0)
typedef unsigned long long ll;
using namespace std;

int main() {
    fastio;
    int n; cin >> n;
    
    ll fibo[46];
    fibo[0] = 0; fibo[1] = 1;
    for (int i = 2; i <= n; ++i) fibo[i] = fibo[i - 1] + fibo[i - 2];
    cout << fibo[n];
}

File: _boj_all/2747.py
dp = [0] * 46
def fibo(n):
    if n==0:
        return 0
    if n==1 or n==2 : 
        return 1
    if dp[n] != 0:
        return dp[n]
    dp[n] = fibo(n-1) + fibo(n-2)
    return dp[n]
print(fibo(int(input())))